### ISSUE TITLE
The cfssl container run in hack/local-up-cluster.sh gets permissions errors

### ISSUE THREAD

**Issue by jeffvance** [kubernetes/kubernetes#38165]
Get various config errors after rebasing to latest k8s and running `local-up-cluster.sh`:
```
Failed to load config file: {"code":5200,"message":"could not read configuration file"}Failed to parse input: unexpected end of JSON input
mv: cannot stat '/var/run/kubernetes/client-kubelet-key.pem': No such file or directory
Failed to load config file: {"code":5200,"message":"could not read configuration file"}Failed to parse input: unexpected end of JSON input
mv: cannot stat '/var/run/kubernetes/client-kube-proxy-key.pem': No such file or directory
Failed to load config file: {"code":5200,"message":"could not read configuration file"}Failed to parse input: unexpected end of JSON input
mv: cannot stat '/var/run/kubernetes/client-controller-key.pem': No such file or directory
Failed to load config file: {"code":5200,"message":"could not read configuration file"}Failed to parse input: unexpected end of JSON input
mv: cannot stat '/var/run/kubernetes/client-scheduler-key.pem': No such file or directory
```
with the end result being that `kubectl get nodes` shows no resources.

Version:
```
Client Version: version.Info{Major:"1", Minor:"4", GitVersion:"v1.4.0+776c994",
GitCommit:"776c994", GitTreeState:"clean", BuildDate:"2016-11-03T21:47:56Z",
GoVersion:"go1.6.1", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"6+",
GitVersion:"v1.6.0-alpha.0.1439+608df651db3545-dirty",
GitCommit:"608df651db3545b1171f424fa52e245a1449c807", GitTreeState:"dirty",
BuildDate:"2016-12-05T21:20:10Z", GoVersion:"go1.6.1", Compiler:"gc",
Platform:"linux/amd64"}
```

`hack/local-up-cluster.sh` function create_client_certkey() runs the cfssl container bind-mounted to /var/run/kubernetes.  Shelling into this container you'll see there are no `owner:group` for the files in its /cert dirt (which is the host's /var/run/kubernetes dir):
```
root@a5bfa7d640f9:/certs# ls -l    #pwd is /cert
ls: cannot access client-ca.crt: Permission denied
ls: cannot access client-ca.key: Permission denied
ls: cannot access auth-proxy-client-ca.crt: Permission denied
ls: cannot access auth-proxy-client-ca.key: Permission denied
ls: cannot access scheduler.kubeconfig: Permission denied
ls: cannot access controller.kubeconfig: Permission denied
ls: cannot access kube-proxy.kubeconfig: Permission denied
ls: cannot access kubelet.kubeconfig: Permission denied
ls: cannot access apiserver.key: Permission denied
ls: cannot access apiserver.crt: Permission denied
ls: cannot access client-ca-bundle.crt: Permission denied
ls: cannot access auth-proxy-client-ca-config.json: Permission denied
ls: cannot access client-ca-config.json: Permission denied
total 0
-????????? ? ? ? ?            ? apiserver.crt
-????????? ? ? ? ?            ? apiserver.key
-????????? ? ? ? ?            ? auth-proxy-client-ca-config.json
-????????? ? ? ? ?            ? auth-proxy-client-ca.crt
-????????? ? ? ? ?            ? auth-proxy-client-ca.key
-????????? ? ? ? ?            ? client-ca-bundle.crt
-????????? ? ? ? ?            ? client-ca-config.json
-????????? ? ? ? ?            ? client-ca.crt
-????????? ? ? ? ?            ? client-ca.key
-????????? ? ? ? ?            ? controller.kubeconfig
-????????? ? ? ? ?            ? kube-proxy.kubeconfig
-????????? ? ? ? ?            ? kubelet.kubeconfig
-????????? ? ? ? ?            ? scheduler.kubeconfig
root@a5bfa7d640f9:/certs# cat client-ca-config.json
cat: client-ca-config.json: Permission denied
```
If I run the cfssl container with `--privileged` it all works both when shelling in (root is owner and group), and `kubectl get nodes` works as expected.

Q: is there something wrong with my local env on the latest kube? Or, should the cfssl container be run privileged now? Or is there a smaller hammer?

**Comment by jsafrane** [comment]
SELinux was culprit in my case (Fedora 24), containers can't write to /var/run. I am not sure what's official way around this except disabling it.

**Comment by jeffvance** [comment]
Thanks @jsafrane. I'm running kube in rhel vms. However, the cfssl container is not fedora based is it?
From cfssl container:
```
# uname -a
Linux b0db5531b84c 3.10.0-327.4.5.el7.x86_64 #1 SMP Thu Jan 21 04:10:29 EST 2016 x86_64 GNU/Linux
root@b0db5531b84c:/certs# getenforce
bash: getenforce: command not found
```

**Comment by jsafrane** [comment]
It does not matter what's inside the container, SELinux on the host blocks writes to /var/lib

**Comment by jsafrane** [comment]
well... host... probably a VM in your case. The thing that runs Docker.

**Comment by jeffvance** [comment]
ok that makes sense.
`setenforce 0` on host vm fixes the permission problem. No need to change local-up-cluster.sh.

**Comment by sjug** [comment]
I think @sttts wrote the new cert parts, looping him in.

**Comment by jeffvance** [comment]
@sjug Here is the local-up-cluster cmd I've been using for months:
```
KUBERNETES_PROVIDER=local LOG_LEVEL=5 ALLOW_SECURITY_CONTEXT=true ALLOW_PRIVILEGED=true hack/local-up-cluster.sh
```
With the latest k8s this command fails with the errors I've listed above. I'm running it on a rhel 7.2 vm and selinux was enforcing.

**Comment by sttts** [comment]
Here is the PR fixing the selinux context issue: https://github.com/kubernetes/kubernetes/pull/37985

**Comment by jeffvance** [comment]
closed by #37985
